include/dlaf/memory/memory_chunk.h

```diff
@@ -131,18 +127,15 @@ class MemoryChunk {
 private:
   void deallocate() {
     if (allocated_) {
-#ifdef DLAF_WITH_CUDA
       if (device == Device::CPU) {
-        DLAF_CUDA_CALL(cudaFreeHost(ptr_));
+        // FIXME: Pinned host memory DLAF_CUDA_CALL(cudaFreeHost(ptr_));
```

Comment:
Yes! This is a must eventually. Has there been any discussion about passing something context/executor/device-like explicitly to the various solvers and copy?

Reply:
I've removed these changes from this PR.